rllib/core/learner/learner_group.py

```diff
@@ -557,6 +551,197 @@ def _load_state(w):
+            for module_id in modules_to_load:
+                if not isinstance(module_id, str):
+                    raise ValueError("modules_to_load must be a list of strings.")
+
+        if self.is_local:
+            module_keys = set(self._learner.module.keys())
+        else:
+            workers = self._worker_manager.healthy_actor_ids()
+            module_keys = set(
+                self._get_results(
+                    self._worker_manager.foreach_actor(
+                        lambda w: w.module.keys(), remote_actor_ids=[workers[0]]
+                    )
+                )[0]
+            )
+
+        if marl_module_ckpt_dir and rl_module_ckpt_dirs:
+            # If both a MARLModule checkpoint and RLModule checkpoints are specified,
+            # the RLModule checkpoints take precedence over the MARLModule checkpoint,
+            # so we should not load any modules in the MARLModule checkpoint that are
+            # also in the RLModule checkpoints.
+            if modules_to_load:
+                for module_id in rl_module_ckpt_dirs.keys():
+                    if module_id in modules_to_load:
+                        raise ValueError(
+                            f"module_id {module_id} was specified in both "
+                            "modules_to_load and rl_module_ckpt_dirs. Please only "
+                            "specify a module to be loaded only once, either in "
+                            "modules_to_load or rl_module_ckpt_dirs, but not both."
+                        )
+            else:
+                modules_to_load = module_keys - set(rl_module_ckpt_dirs.keys())
+
+        # No need to do any file transfer operations if we are running training
+        # on the experiment head node.
+        if self._is_local:
+            if marl_module_ckpt_dir and rl_module_ckpt_dirs:
+                # If both a MARLModule checkpoint and RLModule checkpoints are
+                # specified, load the MARLModule checkpoint first and then load
+                # the RLModule checkpoints.
+                self._learner.module.load_state(
+                    marl_module_ckpt_dir, modules_to_load=set(modules_to_load)
+                )
+                for module_id, path in rl_module_ckpt_dirs.items():
+                    self._learner.module[module_id].load_state(
+                        path / RLMODULE_STATE_DIR_NAME
+                    )
+
+            elif marl_module_ckpt_dir:
+                self._learner.module.load_state(
+                    marl_module_ckpt_dir, modules_to_load=modules_to_load
+                )
+            else:
+                for module_id, path in rl_module_ckpt_dirs.items():
+                    self._learner.module[module_id].load_state(
+                        path / RLMODULE_STATE_DIR_NAME
+                    )
+        else:
+            assert len(self._workers) == self._worker_manager.num_healthy_actors()
```

Comment:
should we write the else logics in a separate util function?

Reply:
yeah lemme go ahead and do that this function has gotten too big.